### PR TITLE
Daily precipitation tables

### PR DESCRIPTION
**Pull request by eric-gade**
## What does this PR do? 🛠️
This PR adds precipitation tables to each of the daily hourly details (if data is available for that day). It addresses #1031. 

## What does the reviewer need to know? 🤔
We have removed the `HourlyPrecipitationBlock` and any references to it, since this construct no longer makes sense as its own block. It is now a constituent part of the `DailyForecastBlock`.

## Screenshots (if appropriate): 📸

<details>
<summary>New precip details:</summary>

Screen Shot 2024-04-24 at 10 25 50 AM

</details>

### CODE REVIEW

**Comment by greg-does-weather** [comment]
For visibility, this PR is like 90% there, but we discovered some funky behavior regarding "today"

**Comment by colinmurphy01** [comment]
What's the status on this? Is it possible for this to be included in Wednesday's release?

**Comment by colinmurphy01** [comment]
Whoa. Actions speak louder than words.

File: web/themes/new_weather_theme/templates/partials/daily-forecast-list-item.html.twig [hunk 1]
```diff
@@ -68,21 +68,24 @@ only be a nighttime period.
       </div>
       {% endfor %}
     </div>
-    <wx-hourly-toggle class="text-primary margin-top-1" role="button" aria-expanded="false" aria-controls="{{itemId}}-hourly-table" data-expanded-text="{{ 'Hide hourly details' | t}}" data-hidden-text="{{'Show hourly details' | t}}" data-expanded-icon-name="expand_less" data-hidden-icon-name="expand_more" tabindex="0">
+    <wx-hourly-toggle class="text-primary-dark margin-top-1" role="button" aria-expanded="false" aria-controls="{{itemId}}-hourly-togglee" data-expanded-text="{{ 'Hide hourly details' | t}}" data-hidden-text="{{'Show hourly details' | t}}" data-expanded-icon-name="expand_less" data-hidden-icon-name="expand_more" tabindex="0">
```
Comment (greg-does-weather) [inline web/themes/new_weather_theme/templates/partials/daily-forecast-list-item.html.twig:71]: This is twice now I've seen `togglee` here and thought it was a typo. But nope. 😂

File: web/modules/weather_data/src/Service/Test/FilterHourlyPrecipitationToDay.php.test [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        $utility = new PrecipAmountMock();
+        $actual = $utility->filterHourlyPrecipitationToDay(
+            $today,
+            [$overlapInput],
+            null,
+        );
+
+        $this->assertEquals($expected, $actual);
+    }
+}
```
Comment (greg-does-weather) [inline web/modules/weather_data/src/Service/Test/FilterHourlyPrecipitationToDay.php.test:134]: These tests are so easy to read.

File: web/themes/new_weather_theme/templates/block/block--weathergov-daily-forecast.html.twig [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
-    <ol class="usa-list--unstyled">
+    <div class="grid-container padding-x-0 tablet:padding-x-2">
+      <ol class="usa-list--unstyled">
       {# Attempt to render the period(s) associated with the current day #}
-      {% if content.today %}
-      {% include '@new_weather_theme/partials/daily-forecast-list-item.html.twig' with { 'periods': content.today, 'dayHours': content.todayHourly, 'alertPeriods': content.todayAlerts, 'dayLabel': "Today" | t } %}
+        {% if content.today %}
+          {% set precipHours = content.precipitationPeriods[0] %}
+          {% include '@new_weather_theme/partials/daily-forecast-list-item.html.twig' with { 'periods': content.today, 'dayHours': content.todayHourly, 'alertPeriods': content.todayAlerts, 'precipHours': precipHours, 'dayLabel': "Today" | t } %}
+          {% set precipIndex = 1 %}
```
Comment (greg-does-weather) [inline web/themes/new_weather_theme/templates/block/block--weathergov-daily-forecast.html.twig:14]: I'll fetch this locally and play with it tomorrow, but looking at it tonight, I can't see where `precipIndex` ever gets incremented.

File: web/modules/weather_data/src/Service/DailyForecastTrait.php [hunk 4]
```diff
@@ -196,11 +197,50 @@ public function getDailyForecastFromGrid(
...
+        $precipPeriods = array_map(function ($startTime) use (
+            &$precipPeriods,
+            &$timezone,
+        ) {
+            return $this->filterHourlyPrecipitationToDay(
+                $startTime,
+                $precipPeriods,
+                $timezone,
+            );
+        }, $periodStartTimes);
```
Comment (greg-does-weather) [inline web/modules/weather_data/src/Service/DailyForecastTrait.php:236]: This all seems like it should work! The data structure is a little confusing, but that's simply _true_, not an implementation issue.

**Review by greg-does-weather — COMMENTED** [review]
I don't see any code issues. Will approve it once I pull it down and check it out locally tomorrow morning!

File: web/themes/new_weather_theme/templates/block/block--weathergov-daily-forecast.html.twig (hunk 3 above)
Comment (eric-gade) [reply web/themes/new_weather_theme/templates/block/block--weathergov-daily-forecast.html.twig:14]: line 19. We add the index of the current loop to the precipIndex. Probably poorly named, should be "offset" or something like that

File: web/themes/new_weather_theme/templates/partials/daily-forecast-list-item.html.twig (hunk 1 above)
Comment (eric-gade) [reply web/themes/new_weather_theme/templates/partials/daily-forecast-list-item.html.twig:71]: Maybe we change to `toggle-target`?

Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/partials/daily-forecast-list-item.html.twig:71]: Naw, I think it's fine. I laughed both times.

File: web/themes/new_weather_theme/templates/block/block--weathergov-daily-forecast.html.twig (hunk 3 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/block/block--weathergov-daily-forecast.html.twig:14]: Ooooh, I see now. I was expected `precipIndex` to get incremented over time, but that's not its purpose. It's there to advance the index past the "today" precip table while iterating through the "detailed" periods. Got it now!

**Review by greg-does-weather — APPROVED** [review]
singing in the rain

**Pull request merged**